pygatt/backends/bgapi/bgapi.py

```diff
@@ -127,19 +127,43 @@ def start(self):
 
         self._ser = serial.Serial(self._serial_port, baudrate=115200,
                                   timeout=0.25)
+
+        # Blow everything away and start anew.
+        # Only way to be sure is to burn it down and start again.
+        # (Aka reset remote state machine)
+        # Note: Could make this a conditional based on parameter if this
+        # happens to be too slow on some systems.
+
+        # The zero param just means we want to do a normal restart instead of
+        # starting a firmware update restart.
+        self.send_command(CommandBuilder.system_reset(0))
+
+        self._ser.flush()
+        self._ser.close()
+        self._ser = None
+        while self._ser is None:
+            try:
+                self._ser = serial.Serial(self._serial_port, baudrate=115200,
```

Comment:
Since this is the same as line 128, could you pull it up to a small function? It probably wouldn't hurt to have the same retry logic for the initial connection, either.